xscen/ensembles.py

```diff
@@ -62,38 +76,157 @@ def ensemble_stats(
             create_kwargs["engine"] = "zarr"
 
     ens = ensembles.create_ensemble(datasets, **create_kwargs)
-    ens_stats = getattr(ensembles, statistics)(ens, **stats_kwargs)
+
+    ens_stats = xr.Dataset(attrs=ens.attrs)
+    for stat in statistics.keys():
+        stats_kwargs = deepcopy(statistics.get(stat, None) or {})
+        if (
+            weights is not None
+            and "weights" in inspect.getfullargspec(getattr(ensembles, stat))[0]
+        ):
+            stats_kwargs["weights"] = weights.reindex_like(ens.realization)
+        if "ref" in stats_kwargs:
+            stats_kwargs["ref"] = ens.loc[stats_kwargs["ref"]]
+
+        if stat == "change_significance":
+            for v in ens.data_vars:
+                with xc.set_options(keep_attrs=True):
```

Comment:
```suggestion
                with xr.set_options(keep_attrs=True):
```
?
Le `keep_attrs=True` de `xclim` n'a un impact que sur les indicateurs pour le moment.